### PR TITLE
Merge release-10.2.1 back to stable10

### PR DESCRIPTION
**Pull request by phil-davis**
## Description
Merge of `release-10.2.1` branch back to `stable10`

I did something like:
```
git checkout stable10
git checkout -b mystable10      ! to get a dummy branch to merge to
git merge release-10.2.1
```
and then resolve all the conflicts it told me about.
There was nothing difficult to resolve. Mostly it was:

- silly conflicts down the end of `l10n` files related to lines being added, and the previous line at the of the array gets a comma added to it, causing a conflict to be reported.
- conflicts in acceptance test files because a few acceptance test changes have happened in `stable10` even in the last few days, that were not in `release-10.2.1` and were messy enough that they caused conflicts. Taking all the newer code already in `stable10` was the thing to do.

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Database schema changes (next release will require increase of minor version instead of patch)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests only (no source changes)

## Checklist:
- [ ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

File: apps/files_sharing/l10n/pt_BR.js [hunk 1]
```diff
@@ -137,5 +137,10 @@ OC.L10N.register(
     "Nothing to configure." : "Nada para configurar.",
     "You unshared %1$s shared by %2$s from self" : "Você descompartilhou %1$s compartilhado por %2$s de si mesmo",
     "Unshared %1$s from self" : "Descompartilhou %1$s de sí mesmo"
+=======
```
Comment (phil-davis) [inline apps/files_sharing/l10n/pt_BR.js]: Note to self: I stuffed this one up

File: apps/files_sharing/tests/Controller/Share20OcsControllerTest.php [hunk 2]
```diff
@@ -2288,6 +2288,7 @@ public function testUpdateOtherPermissions() {
 
 	public function testUpdateShareCannotIncreasePermissions() {
 		$ocs = $this->mockFormatShare();
+
```
Comment (phil-davis) [inline apps/files_sharing/tests/Controller/Share20OcsControllerTest.php]: Note to self: I can remove this blank line

File: composer.lock [hunk 3]
```diff
@@ -1266,7 +1266,7 @@
             "time": "2015-08-01T16:27:37+00:00"
         },
         {
-            "name": "jeremeamia/SuperClosure",
+            "name": "jeremeamia/superclosure",
```
Comment (phil-davis) [inline composer.lock]: This `composer.lock` stuff comes from needing to do `composer update` to "resolve" "conflicts" reported for `composer.lock`
It is harmless.

File: lib/l10n/de.js [hunk 4]
```diff
@@ -192,7 +192,7 @@ OC.L10N.register(
     "No maximum ownCloud version is defined in appinfo/info.xml." : "In appinfo/info.xml ist keine maximale ownCloud-Version definiert.",
     "Can't create or write into the apps-external directory %s" : "Das \"apps-extern\"-Verzeichnis %s kann nicht erstellt oder es kann darin nicht geschrieben werden.",
     "Can't update %s" : "%s kann nicht aktualisiert werden",
-    "Access to this resource or one of its sub-items has been denied." : "Der Zugriff auf diese Ressource oder eine ihrer Unterelemente wurde verweigert.",
+    "Access to this resource or one of its sub-items has been denied." : "Der Zugriff auf diese Ressource oder eine derer Unterelemente wurde verweigert.",
```
Comment (phil-davis) [inline lib/l10n/de.js]: @pmaier1 is this correct? The changed text here is the improvement?

File: apps/files_sharing/l10n/pt_BR.js (hunk 1 above)
Comment (phil-davis) [reply apps/files_sharing/l10n/pt_BR.js]: fixed

**Comment by phil-davis** [comment]
This was somewhat painful and tedious to resolve all the stupid language file conflicts. But at least the diffs do look reasonable.

The big pain about this PR is that it seems to have listed every PR since the original point at which `release-10.2.0` was branched off from `stable10`. Actually 99% of that is code that is already in `stable10`. So lots of the PRs listed here are a NOOP. It is a pity that the `git merge` process is too stupid to drop the ones that already have an exact match in `stable10`.

Maybe we should take this `mystable10` branch and then squash the commits so we just get a single merge commit?

File: lib/l10n/de.js (hunk 4 above)
Comment (pmaier1) [reply lib/l10n/de.js]: No, this is rather an "imworsement" ^_^. Actually correct translation would be:

"Der Zugriff auf diese Ressource oder eine**s** ihrer Unterelemente wurde verweigert."

Comment (patrickjahns) [reply lib/l10n/de.js]: If we could drop the ln10 changes and just fetch them fresh from transifex it would be good

Comment (phil-davis) [reply lib/l10n/de.js]: Then I will look at the remaining changes here, and make a new PR that just cherry-picks the relevant commits out of `release-10.2.1` branch. We can verify that the diffs are as expected (match what is here). Then we will not need to actually merge this painful PR.

**Comment by phil-davis** [comment]
See #35763 for the small version of this, without the crazy-long list of commits, and without the `l10n` translation changes.

**Comment by phil-davis** [comment]
PR #35763 has been merged to `stable10` so closing this PR. I will leave the branch here in case someone cares to see it.

**Pull request closed without merging**